### PR TITLE
Switch to DemoCards.jl

### PR DESCRIPTION
**Pull request by Dsantra92**
This pr sets DemoCards for the tutorials page, removing direct Pluto and StaicHTML dependency.

Also added cache files for reducing time on CI. With a few more notebooks, running CI on notebooks would be very costly.

Merge after [DemoCards.jl pluto pr](https://github.com/JuliaDocs/DemoCards.jl/pull/121)

Screenshots:
screenshot1
screenshot

### CODE REVIEW

**Comment by CarloLucibello** [comment]
Nice, I'll wait https://github.com/JuliaDocs/DemoCards.jl/pull/121 before testing locally

File: docs/Project.toml [hunk 1]
```diff
@@ -1,13 +1,12 @@
 [deps]
+DemoCards = "311a05b2-6137-4a5a-b473-18580a3d38b5"
```
Comment (CarloLucibello) [inline docs/Project.toml:2]: we need compat bounds here

**Review by CarloLucibello — CHANGES_REQUESTED** [review]

**Comment by CarloLucibello** [comment]
The doc CI is still failing

**Comment by Dsantra92** [comment]
Looking into it shortly.

**Comment by Dsantra92** [comment]
Is the latest version of DemoCards locally available? I think I messed up the release 🙁. Didn't use a tagbot.

**Comment by CarloLucibello** [comment]
see [here](https://github.com/JuliaDocs/DemoCards.jl/commit/68665066201d89a2e95f27489bd42117cc8771a8#commitcomment-85795525)

**Comment by Dsantra92** [comment]
The docs are failing because I changed the pluto files themselves, working on it.

**Comment by Dsantra92** [comment]
Managing rendered files can be a pain. Adding a CRON job later for running the files and saving them (committed by some GitHub bot) after some weeks make sense.

**Comment by Dsantra92** [comment]
Due to local rendering the pages should have effect of https://github.com/JuliaDocs/DemoCards.jl/pull/126. So no blocking pr.

**Comment by CarloLucibello** [comment]
The markdown outputs shouldn't be committed, right?

**Comment by CarloLucibello** [comment]
ok, let's keep them. So this is good to go?

**Comment by Dsantra92** [comment]
Just check things again locally on your machine and we are good to go.

**Comment by CarloLucibello** [comment]
I tested it locally and the notebooks look fine. The tutorial landing page looks like this:
Screenshot 2022-10-07 at 05 00 36

The double "Tutorials" header is not nice, let's avoid the subsection and let's just write a few introductory lines also inviting people to contribute, e.g. something like "You will find below some tutorials showing different features of GraphNeuralNetworks.jl ..... Users are welcome to contribute their own demonstrations". 

Also, it would be nice to have a separate and related image for each card.

**Comment by Dsantra92** [comment]
Preview:
image

**Comment by Dsantra92** [comment]
The documentation does not use the cached md files. I wonder why SHA of the files in my computer and that of the CI doesn't match.

**Comment by Dsantra92** [comment]
Something is acting weird with the pluto notebooks cache. Let's remove that for the time being and merge?

**Comment by CarloLucibello** [comment]
> Something is acting weird with the pluto notebooks cache. Let's remove that for the time being and merge?

ok

**Comment by Dsantra92** [comment]
Ready to merge

**Pull request merged**